graphql/src/main/java/com/scalar/db/graphql/server/ServerConfig.java

```diff
@@ -0,0 +1,110 @@
+package com.scalar.db.graphql.server;
+
+import com.google.common.base.Strings;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.LinkedHashSet;
+import java.util.Properties;
+import java.util.Set;
+import javax.annotation.concurrent.Immutable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Immutable
+@SuppressFBWarnings("JCIP_FIELD_ISNT_FINAL_IN_IMMUTABLE_CLASS")
+class ServerConfig {
+  public static final String PREFIX = "scalar.db.graphql.";
+  public static final String PORT = PREFIX + "port";
+  public static final String PATH = PREFIX + "path";
+  public static final String NAMESPACES = PREFIX + "namespaces";
+  public static final String GRAPHIQL = PREFIX + "graphiql";
+
+  public static final int DEFAULT_PORT = 8080;
+  public static final boolean DEFAULT_GRAPHIQL = true;
+  private static final String DEFAULT_PATH = "/graphql";
+  private static final Logger LOGGER = LoggerFactory.getLogger(ServerConfig.class);
+  private final Properties props;
+  private int port;
+  private String path;
+  private Set<String> namespaces;
+  private boolean graphiql;
+
+  public ServerConfig(File propertiesFile) throws IOException {
+    this(new FileInputStream(propertiesFile));
+  }
+
+  public ServerConfig(InputStream stream) throws IOException {
+    props = new Properties();
+    props.load(stream);
+    load();
+  }
+
+  public ServerConfig(Properties properties) {
+    props = new Properties();
+    props.putAll(properties);
+    load();
+  }
+
+  private void load() {
+    port = getInt(PORT, DEFAULT_PORT);
+    path = props.getProperty(PATH, DEFAULT_PATH);
+    namespaces = new LinkedHashSet<>(Arrays.asList(getStringArray(NAMESPACES)));
+    graphiql = getBoolean(GRAPHIQL, DEFAULT_GRAPHIQL);
+  }
+
+  private int getInt(String name, int defaultValue) {
+    String value = props.getProperty(name);
+    if (Strings.isNullOrEmpty(value)) {
+      return defaultValue;
+    }
+    try {
+      return Integer.parseInt(value);
+    } catch (NumberFormatException ignored) {
+      LOGGER.warn(
+          "the specified value of '{}' is not a number. using the default value: {}",
+          name,
+          defaultValue);
+      return defaultValue;
+    }
+  }
+
+  private String[] getStringArray(String name) {
```

Comment:
You can also add this `getStringArray` method to `ConfigUtils` and use it.

Reply:
@brfrn169 @feeblefakie 
Question: Should we generally allow spaces around `,` characters when we specify a list of strings in a properties file? I did so here using a regexp `"\\s*,\\s*"` https://github.com/scalar-labs/scalardb/pull/439/commits/6ad93d979cb7c278469186e1aaeb658ec1481b22#diff-690e1302005478e3eb87bc31ffafa5aeb72308567a9927a7e961d7ef6cf09871R60 .
If it's acceptable I'll update [`DatabaseConfig`](https://github.com/scalar-labs/scalardb/blob/270437a9d412954972869b69843b2e3642c5811c/core/src/main/java/com/scalar/db/config/DatabaseConfig.java#L120) to use `ConfigUtils.getStringArray` later on.

Reply:
I think using the regexp `"\\s*,\\s*"` to split the comma separated string is fine.